### ISSUE TITLE
[API] Add image url's to the content block response

### ISSUE THREAD

**Issue by matsantosz** [statikbe/laravel-filament-flexible-content-blocks#48]
I suppose this plugin is meant to be used in a monolithic laravel application, it even provides the blade components to be used in your page.

But i am actually using it as a CMS only and have an api endpoint to fetch the articles.

The problem is when using image blocks, we don't store the image url's inside the content_blocks column, meaning the front end will not be aware of the image url, only the block_id.

I implemented a workaround where i have to iterate over all the blocks and check for images, then get the specific image url, but that seems a little too much, is there an easy way for the image_url to be present in the json response?

---

Also, can i completely disable image conversion and responsive images?

**Comment by sten** [comment]
@matsantosz That is an interesting idea. 

We have not yet used the package through an API. But it should be easy to add a trait for use in an API resource class, to output proper JSON of the content blocks with image UUIDs or URLs. I would see it as a trait with a function that has the record as an argument, that would output an array of the `content_blocks` var. What do you think?

With regard of disabling all image conversions, I don't think that is possible right now. It could be done by adding a boolean config parameter and adding a condition around each image conversion code section. PR is welcome!

**Comment by matsantosz** [comment]
Hey @sten,

I like your idea, but that would essentially be the same workaround as i did, no? Take a look at how i wrote it:
![image](https://github.com/user-attachments/assets/19573f4f-466d-4599-af58-039c0e265d79)

There's probably a better way of doing this, but what do you think of this solution, if that's possible:

We have an interface implemented in model, that would specify to the application that we are using it for API purposes, we intercept it upon saving and add image_url inside the block.

I'm not sure if it makes sense to return the responsive images as well.

Which one do you vote?

**Comment by sten** [comment]
@matsantosz Thanks for the code sample. 

I would put the toArray logic in each block, so each block can handle its own array format. Then in the API resource I would resolve the content blocks and iterate over them. This way you do not have to extend your function if there are other blocks with images or something else later on.

Your proposal of changing the filament builder data on save, I find a bit problematic:
1. the data is no longer normalised. You are storing the same data twice.
2. the builder form component manages this data, so overwriting this has led me in the past to tricky errors ;-(.

**Comment by matsantosz** [comment]
I really like this idea @sten, i must ask, how would you resolve the content blocks? there might be some helper method that don't know about for resolving the data into blocks 😸

**Comment by sten** [comment]
@matsantosz Yes, you can do that. The `getSearchableBlockContent` function in https://github.com/statikbe/laravel-filament-flexible-content-blocks/blob/main/src/Models/Concerns/HasContentBlocksTrait.php does something similar.

It should be a matter of hitting `getMedia(['block' => $block->block_id])` and adding it to the data array that you want.

**Comment by matsantosz** [comment]
Thank you @sten! I will provide the code based on your suggestion, hopefully it helps someone:

First, i created a simple interface with a toArray method:
```php
namespace App\ContentBlocks\Contracts;

interface FlexibleBlock
{
    public function toArray(): array;
}
```

Now, every block should implement this interface and return `$this->blockData`, see the example for TextBlock:
```php
namespace App\ContentBlocks;

class TextBlock extends BaseTextBlock implements FlexibleBlock
{
    public function toArray(): array
    {
        return $this->blockData;
    }
}
```

For image blocks or other blocks with media uploads, you may customize the toArray implementation like so:
```php
class ImageBlock extends BaseImageBlock implements FlexibleBlock
{
    public function toArray(): array
    {
        return [
            ...$this->blockData,
            'image_url' => $this->getMediaUrl($this->getBlockId()),
        ];
    }
}
```

Now, every block should have its implementation of the toArray method, we may use it on our API resource:
```php
public function toArray(Request $request): array
{
    $contentBlocks = new ContentBlocks($this->resource);
    $content = collect($contentBlocks->contentBlocks)->map(fn (FlexibleBlock $block) => $block->toArray());

    return [
        'content' => $content,
    ];
}
```

Thanks for the help and for this awesome plugin @sten!
